src/shared/components/FluxMonacoEditor.tsx

```diff
@@ -34,41 +29,22 @@ export interface EditorProps {
   wrapLines?: 'off' | 'on' | 'bounded'
 }
 
-interface Props extends EditorProps {
-  setEditorInstance?: (editor: EditorType) => void
-}
-
-const FluxEditorMonaco: FC<Props> = ({
+const FluxEditorMonaco: FC<EditorProps> = ({
   script,
   onChangeScript,
   onSubmitScript,
-  setEditorInstance,
   autogrow,
   readOnly,
   wrapLines,
 }) => {
   const lspServer = useRef<LSPServer>(null)
-  const [editorInst, seteditorInst] = useState<EditorType | null>(null)
   const [docURI, setDocURI] = useState('')
 
   const wrapperClassName = classnames('flux-editor--monaco', {
     'flux-editor--monaco__autogrow': autogrow,
   })
 
-  const updateDiagnostics = (diagnostics: Diagnostic[]) => {
-    if (editorInst) {
-      const results = p2m.asDiagnostics(diagnostics)
-      monacoEditor.setModelMarkers(editorInst.getModel(), 'default', results)
-    }
-  }
-
```

Comment:
we no longer need to handle setting the model markers in the React component - the `monaco.flux.server` handles this for us when it receives a message via `onMessage` and updates the appropriate monaco instance via the URI in the message